src/nvim/ex_cmds2.c

```diff
@@ -447,6 +447,19 @@ int buf_write_all(buf_T *buf, bool forceit)
   int retval = (buf_write(buf, buf->b_ffname, buf->b_fname,
                           1, buf->b_ml.ml_line_count, NULL,
                           false, forceit, true, false));
+
+  if (retval == OK && has_event(EVENT_BUFMODIFIEDSET) && buf->b_changed_invalid) {
+    bufref_T bufref;
+    set_bufref(&bufref, buf);
+    aco_save_T aco;
+    aucmd_prepbuf(&aco, buf);
+    apply_autocmds(EVENT_BUFMODIFIEDSET, NULL, NULL, false, buf);
```

Comment:
since crashes are already being reported, perhaps we should consider deferring this event? i.e. https://github.com/neovim/neovim/issues/35366

Reply:
you mean `set charconvert=execute('%bw!')` this crash ? I don’t think so, using bwipe in charconvert is weird

Reply:
sure but weird things happen, and crashes should never happen. also, it would avoid needing to fiddle with `b_locked`, which is basically a hack.